### DISCUSSION TITLE
Access-Control-Allow-Origin CORS errors due to incorrect owner on Shlink directory

### DISCUSSION THREAD

**Discussion by sosojustdo** [shlinkio/discussions#130] [Help wanted] [answered]
### Shlink version

3.6.4

### PHP version

8.1.10

### How do you serve Shlink

Self-hosted nginx

### Database engine

MySQL

### Database version

5.7.32

### Summary

Add server to https://app.shlink.io, The browser console shows an error
<img width="1675" alt="image" src="https://github.com/shlinkio/shlink/assets/3364336/9d29a787-05e5-421f-8bf1-2ba564800c13">

curl -X OPTIONS https://www.example.com/rest/v3/health -H  "accept: application/json" -H "origin: https://app.shlink.io" -H "Access-Control-Request-Headers: content-type,x-api-key" -I  command info:
<img width="1328" alt="image" src="https://github.com/shlinkio/shlink/assets/3364336/a8c4307f-63ea-4e8a-9eed-8c0ed5949a2b">

nginx.conf
<img width="857" alt="image" src="https://github.com/shlinkio/shlink/assets/3364336/a1643057-bccf-47d9-8854-8e9d57ba822b">

**Comment by acelaya** [comment]
By the 500 error on the second line, I'm guessing there's some other issue.

Try the same cURL but with GET.

Also, Shlink generates those headers automatically. You can just let it do it. No need to define them yourself in nginx config.

**Reply by sosojustdo** [reply]
<img width="1155" alt="image" src="https://github.com/shlinkio/shlink/assets/3364336/68f4b60d-6953-4033-b4a4-563a868a3d3d">
@acelaya This curl GET result, how can see the shlink server error log, /data/log dir empty log file, please help me check again, thank you
<img width="524" alt="image" src="https://github.com/shlinkio/shlink/assets/3364336/2479d6d0-bdcf-4240-8159-664c7c1d9471">

**Reply by acelaya** [reply]
Check nginx logs. You should probably find something there.

If those are empty or have no useful information, make sure you followed the steps here https://shlink.io/documentation/install-dist-file/#installation-steps

Pay special attention to the one instructing you should grant write permissions to the data directory.

**Reply by sosojustdo** [reply]
<img width="1180" alt="image" src="https://github.com/shlinkio/shlink/assets/3364336/1e896f1b-1094-47c2-99b5-34ca4546e49c">
Thank you again reply, yes I found nginx error log, can't write shlink log, php-fpm start by nginx user, but shlink root dir is root user.
<img width="659" alt="image" src="https://github.com/shlinkio/shlink/assets/3364336/e75fbb17-f167-4727-a1db-45d81a3dff45">
<img width="450" alt="image" src="https://github.com/shlinkio/shlink/assets/3364336/f1db248c-ad39-4590-882a-cc629d216d90">

**Comment by sosojustdo** [comment, marked as answer]
Finally, after I saved the nginx, php-fpm, and shlink directories as the nginx user and user group, the startup was successful. Thank you for your reply, have nice day
